src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1980,12 +1980,14 @@ public Map<String, String> getTokenToEndpointMap()
 
     public String getLocalHostId()
     {
-        return getTokenMetadata().getHostId(FBUtilities.getBroadcastAddress()).toString();
+        UUID id = getLocalHostUUID();
+        return id != null ? id.toString() : null;
     }
 
     public UUID getLocalHostUUID()
     {
-        return getTokenMetadata().getHostId(FBUtilities.getBroadcastAddress());
+        UUID id = getTokenMetadata().getHostId(FBUtilities.getBroadcastAddress());
```

Comment:
what is the probability that id returned by TokenMetadata and id returned by looking at disk would be different?